compiler/src/java_plugin/cpp/java_generator.cpp

```diff
@@ -639,10 +639,13 @@ static void PrintService(const ServiceDescriptor* service,
                          Printer* p,
                          bool generate_nano) {
   (*vars)["service_name"] = service->name();
+  (*vars)["file_name"] = service->file()->name();
   (*vars)["service_class_name"] = ServiceClassName(service);
   p->Print(
       *vars,
-      "@$Generated$(\"by gRPC proto compiler\")\n"
+      "@$Generated$(\n"
+      "    value = \"by gRPC proto compiler\",\n"
+      "    comments = \"$file_name$\")\n"
```

Comment:
I would give the comments some structure, e.g., "Source: $file_name$". This would allow us to add more contents to the comments in the future.